File: ngPortfolio/src/app/model/project.ts
import { Candidate } from "./candidate";

export class Project {
    id: number;
    url: String;
    about: String;
    photoURL: String;
    candidate: Candidate;

    constructor(
        id: number = 0,
        url: String = '',
        about: String = '',
        photoURL: String = '',
        candidate: Candidate = new Candidate()
    )

    {
        this.id = id;
        this.url = url;
        this.about = about;
        this.photoURL = photoURL;
        this.candidate = candidate;
    }
}

File: portfolio/src/main/java/com/alextadkins/portfolio/model/Project.java
package com.alextadkins.portfolio.model;

import com.fasterxml.jackson.annotation.JsonIgnore;

import javax.persistence.*;

@Entity
public class Project {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private int id;
    private String url;
    private String about;
    @Column(name = "photo_url")
    private String photoURL;
    @JsonIgnore
    @ManyToOne
    @JoinColumn(name = "candidate_id")
    private Candidate candidate;

    Project() {}

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getUrl() {
        return url;
    }

    public void setUrl(String url) {
        this.url = url;
    }

    public String getAbout() {
        return about;
    }

    public void setAbout(String about) {
        this.about = about;
    }

    public String getPhotoURL() {
        return photoURL;
    }

    public void setPhotoURL(String photoURL) {
        this.photoURL = photoURL;
    }

    public Candidate getCandidate() {
        return candidate;
    }

    public void setCandidate(Candidate candidate) {
        this.candidate = candidate;
    }
}
